internal/webhooks/capp_mutator_webhook.go

```diff
@@ -0,0 +1,44 @@
+package webhooks
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	cappv1alpha1 "github.com/dana-team/container-app-operator/api/v1alpha1"
+
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+type CappMutator struct {
+	Decoder *admission.Decoder
+}
+
+// +kubebuilder:webhook:path=/mutate-capp,mutating=true,sideEffects=NoneOnDryRun,failurePolicy=fail,groups=rcs.dana.io,resources=capps,verbs=create;update,versions=v1alpha1,name=capp.dana.io,admissionReviewVersions=v1;v1beta1
+
+const MutatorServingPath = "/mutate-capp"
+
+// Handle implements the mutation webhook.
+func (c *CappMutator) Handle(ctx context.Context, req admission.Request) admission.Response {
+	logger := log.FromContext(ctx).WithValues("mutation webhook", "capp mutation Webhook", "Name", req.Name)
+	capp := cappv1alpha1.Capp{}
+	if err := c.Decoder.DecodeRaw(req.Object, &capp); err != nil {
+		logger.Error(err, "could not decode capp object")
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	// Check if annotations already exists, if not create it.
+	if capp.ObjectMeta.Annotations == nil {
+		capp.ObjectMeta.Annotations = make(map[string]string)
+	}
+
+	capp.ObjectMeta.Annotations["rcs.dana.io/last-updated-by"] = req.UserInfo.Username
+
+	marshaledCapp, err := json.Marshal(capp)
+	if err != nil {
+		return admission.Errored(http.StatusInternalServerError, err)
+	}
```

Comment:
I generally like separating the logic in mutating webhooks to a different function, which is more easily testable (not with e2e-tests but in integration testing). See this as [an example](https://github.com/dana-team/env-route-ns-mutator/blob/main/internal/webhook/namespace_webhook.go#L38) and here [the test](https://github.com/dana-team/env-route-ns-mutator/blob/main/internal/webhook/namespace_webhook_test.go)